dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,833 @@
+
+[NOTE]
+====
+This feature is not supported for builds using the xref:build_secrets.adoc#using-secrets-custom-strategy[Custom Strategy].
+====
+
+
+[[using-secrets-during-build]]
+== Input Secrets
+
+In some scenarios, build operations require credentials to access dependent
+resources, but it is undesirable for those credentials to be available in the
+final application image produced by the build. You can define _input secrets_
+for this purpose.
+
+For example, when building a NodeJS application, you can set up your private
+mirror for NodeJS modules. In order to download modules from that private
+mirror, you have to supply a custom *_.npmrc_* file for the build that contains
+a URL, user name, and password. For security reasons, you do not want to expose
+your credentials in the application image.
+
+This example describes NodeJS, but you can use the same approach for adding SSL
+certificates into the *_/etc/ssl/certs_* directory, API keys or tokens, license
+files, etc.
+
+[[using-secrets-in-the-buildconfig]]
+=== Adding Input Secrets
+
+To add an input secret to an existing `BuildConfig`:
+
+. Create the secret, if it does not exist:
++
+----
+$ oc secrets new secret-npmrc .npmrc=~/.npmrc
+----
++
+This creates a new secret named *_secret-npmrc_*, which contains the base64
+encoded content of the *_~/.npmrc_* file.
+
+. Add the secret to the `source` section in the existing `BuildConfig`:
++
+[source,yaml]
+----
+source:
+  git:
+    uri: https://github.com/openshift/nodejs-ex.git
+  secrets:
+    - secret:
+        name: secret-npmrc
+  type: Git
+----
+
+To include the secret in a new `BuildConfig`, run the following command:
+
+----
+$ oc new-build \
+    openshift/nodejs-010-centos7~https://github.com/openshift/nodejs-ex.git \
+    --build-secret secret-npmrc
+----
+
+During the build, the *_.npmrc_* file is copied into the directory where the
+source code is located. In {product-title} S2I builder images, this
+is the image working directory, which is set using the `WORKDIR` instruction
+in the *_Dockerfile_*. If you want to specify another directory, add a
+`destinationDir` to the secret definition:
+
+[source,yaml]
+----
+source:
+  git:
+    uri: https://github.com/openshift/nodejs-ex.git
+  secrets:
+    - secret:
+        name: secret-npmrc
+      destinationDir: /etc
+  type: Git
+----
+
+You can also specify the destination directory when creating a new
+`BuildConfig`:
+
+----
+$ oc new-build \
+    openshift/nodejs-010-centos7~https://github.com/openshift/nodejs-ex.git \
+    --build-secret “secret-npmrc:/etc”
+----
+
+In both cases, the *_.npmrc_* file is added to the *_/etc_* directory of the
+build environment. Note that for a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#docker-build[Docker strategy] the destination directory must be a relative path.
+
+[[using-secrets-s2i-strategy]]
+=== Source-to-Image Strategy
+
+When using a `Source` strategy, all defined source secrets are copied to their
```

Comment:
@bparees In this and the next 2 subsections, the instances of "source secrets" should become "input secrets", yes?